# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: servo/components/style/values/specified/transform.rs
# Commit: 473ebe5ec7aa
# Full Hash: 473ebe5ec7aa4f1c5da45ed580fb56258737a210
# Author: Boris Chiou <boris.chiou@gmail.com>
# Date: 2023-08-23 04:21:33
# Regressor Bug: 1819464
# File Overlap Count: 2
# Description:
#   Bug 1819464 - Part 2: Add content-box and stroke-box into transform-box in style. r=emilio
#   
#   Also, add a prefernce for content-box and stroke-box. So we have to
#   define a new type, TransformBox, and let content-box and stroke-box
#   behind the preference.
# ==============================================================================

diff -r 9c22e16e0567 -r 473ebe5ec7aa servo/components/style/values/specified/transform.rs
--- a/servo/components/style/values/specified/transform.rs	Tue Aug 22 19:49:03 2023 +0000
+++ b/servo/components/style/values/specified/transform.rs	Tue Aug 22 19:49:03 2023 +0000
@@ -35,6 +35,49 @@
     Length,
 >;
 
+#[cfg(feature = "gecko")]
+fn all_transform_boxes_are_enabled(_context: &ParserContext) -> bool {
+    static_prefs::pref!("layout.css.transform-box-content-stroke.enabled")
+}
+
+#[cfg(feature = "servo")]
+fn all_transform_boxes_are_enabled(_context: &ParserContext) -> bool {
+    false
+}
+
+/// The specified value of `transform-box`.
+/// https://drafts.csswg.org/css-transforms-1/#transform-box
+// Note: Once we ship everything, we can drop this and just use single_keyword for tranform-box.
+#[allow(missing_docs)]
+#[derive(
+    Animate,
+    Clone,
+    ComputeSquaredDistance,
+    Copy,
+    Debug,
+    Deserialize,
+    MallocSizeOf,
+    Parse,
+    PartialEq,
+    Serialize,
+    SpecifiedValueInfo,
+    ToAnimatedValue,
+    ToComputedValue,
+    ToCss,
+    ToResolvedValue,
+    ToShmem,
+)]
+#[repr(u8)]
+pub enum TransformBox {
+    #[parse(condition = "all_transform_boxes_are_enabled")]
+    ContentBox,
+    BorderBox,
+    FillBox,
+    #[parse(condition = "all_transform_boxes_are_enabled")]
+    StrokeBox,
+    ViewBox,
+}
+
 impl TransformOrigin {
     /// Returns the initial specified value for `transform-origin`.
     #[inline]